test/functional/vimscript/system_spec.lua

```diff
@@ -639,18 +639,25 @@ describe('shell :!', function()
       1
       4
       2]])
-    feed(':4verbose %!sort<cr>')
-    screen:expect{
-      any=[[Executing command: .?Start%-Process sort %-RedirectStandardInput .* %-RedirectStandardOutput .* %-NoNewWindow %-Wait]]
-    }
+    if iswin() then
+      feed(':4verbose %!sort /R<cr>')
+      screen:expect{
+        any=[[Executing command: .?Start%-Process sort %-ArgumentList " /R" %-RedirectStandardInput .* %-RedirectStandardOutput .* %-NoNewWindow %-Wait]]
+      }
+    else
+      feed(':4verbose %!sort -r<cr>')
+      screen:expect{
+        any=[[Executing command: .?Start%-Process sort %-ArgumentList " %-r" %-RedirectStandardInput .* %-RedirectStandardOutput .* %-NoNewWindow %-Wait]]
```

Comment:
Wonder if we need to worry about the leading space. Depends on how powershell sends the args to the process and whether the process is prepared to trim whitespace around args.
```suggestion
        any=[[Executing command: .?Start%-Process sort %-ArgumentList "%-r" %-RedirectStandardInput .* %-RedirectStandardOutput .* %-NoNewWindow %-Wait]]
```

Reply:
Leading spaces should be here. This PR is keeping the spaces in `cmd_args`. The command that gets executed is the one `make_filter_cmd()` returns, so the leading spaces should exist.

Or are you suggesting I trim the spaces around `cmd_args` myself? That seems too much hassle, honestly.

Reply:
we aren't trimming anything, we're just dealing with a consequence of the parsing we're doing. The example in https://github.com/neovim/neovim/issues/19250#issuecomment-1176757468 looks like:

```
Start-Process jq -ArgumentList "--tab . .\locations.json"
                                ^ no space here
```

Reply:
```powershell
Start-Process jq -ArgumentList "--tab . .\locations.json"
                                ^ no space here
```

But there should be space. The user will run this command --

```vim
:!jq --tab . .\locations.json
```

This PR (as of now, before I commit your suggestion below) will convert the ex command above into the following --

```powershell
Start-Process jq -ArgumentList " --tab . .\locations.json"
                                ^ one space here
```

---

The test I have added, using `sort -r` is the same case. I may be confusing your query. Sorry.